[AOTI][refactor] Update MKLDNN ops cpp wrapper support

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #132367

Summary: Set op_overload for MKLDNN ops so that cpp_kernel_name and python_kernel_name are constructed from there. This is an important step towards support those MKLDNN ops in the ABI-compatible mode, because we will need to read schema from op_overload for generating correct fallback op call in C++.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @chauhang

Differential Revision: [D60909798](https://our.internmc.facebook.com/intern/diff/D60909798)

desertfire committed:
[AOTI][refactor] Update MKLDNN ops cpp wrapper support

Summary: Set op_overload for MKLDNN ops so that cpp_kernel_name and python_kernel_name are constructed from there. This is an important step towards support those MKLDNN ops in the ABI-compatible mode.

[ghstack-poisoned]

desertfire committed:
Update on "[AOTI][refactor] Update MKLDNN ops cpp wrapper support"


Summary: Set op_overload for MKLDNN ops so that cpp_kernel_name and python_kernel_name are constructed from there. This is an important step towards support those MKLDNN ops in the ABI-compatible mode.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames chauhang

[ghstack-poisoned]

desertfire committed:
Update on "[AOTI][refactor] Update MKLDNN ops cpp wrapper support"


Summary: Set op_overload for MKLDNN ops so that cpp_kernel_name and python_kernel_name are constructed from there. This is an important step towards support those MKLDNN ops in the ABI-compatible mode.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames chauhang

[ghstack-poisoned]

desertfire committed:
Update on "[AOTI][refactor] Update MKLDNN ops cpp wrapper support"


Summary: Set op_overload for MKLDNN ops so that cpp_kernel_name and python_kernel_name are constructed from there. This is an important step towards support those MKLDNN ops in the ABI-compatible mode.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames chauhang

[ghstack-poisoned]

leslie-fang-intel commented on torch/_inductor/mkldnn_ir.py:
```diff
@@ -238,10 +238,8 @@ def __init__(
             None,
-            python_kernel_name="torch.ops.mkldnn._convolution_pointwise",
-            cpp_kernel_name="mkldnn::_convolution_pointwise",
+            op_overload=torch.ops.mkldnn._convolution_pointwise.default,
```
Does that mean `self.python_kernel_name` and `self.cpp_kernel_name` should be generated by `self.op_overload`?

desertfire replied:
Yes, they will be generated in the base class ExternKernel.

leslie-fang-intel commented on torch/_inductor/mkldnn_ir.py:
```diff
@@ -799,22 +803,39 @@ def __init__(
-        const_args = []
-        const_args.extend(self.codegen_const_args())
+        const_arg_names = [
+            "act_scale",
```
Do we need to force the exact same name of formal parameters?
There are some mis-using of `act_scale` and `x_scale` between here and op registration, will it be an issue? https://github.com/pytorch/pytorch/blob/56f2917bef45c33d7b29b3247dcf751a898186e5/aten/src/ATen/native/quantized/library.cpp#L266

desertfire replied:
It needs to be an exact match. Let me update it.

Ideally we want to declare all Tensor args before scalar ops. We can't change these ops for BC reason, but let's follow the convention for future ops.

desertfire committed:
Update on "[AOTI][refactor] Update MKLDNN ops cpp wrapper support"


Summary: Set op_overload for MKLDNN ops so that cpp_kernel_name and python_kernel_name are constructed from there. This is an important step towards support those MKLDNN ops in the ABI-compatible mode, because we will need to read schema from op_overload for generating correct fallback op call in C++.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames chauhang

[ghstack-poisoned]

desertfire committed:
Update on "[AOTI][refactor] Update MKLDNN ops cpp wrapper support"


Summary: Set op_overload for MKLDNN ops so that cpp_kernel_name and python_kernel_name are constructed from there. This is an important step towards support those MKLDNN ops in the ABI-compatible mode, because we will need to read schema from op_overload for generating correct fallback op call in C++.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames chauhang

[ghstack-poisoned]

desertfire commented:
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D60909798).
